### PR TITLE
Allow for using a short version of the node name

### PR DESCRIPTION
**Pull request by andrewrynhard**
This PR allows users to specify that the node name should be the first part of the EC2 instances' private dns name. For example `ip-10-35-99-104.us-west-1.compute.internal` becomes `ip-10-35-99-104`.

Only when a cluster is configured with the AWS cloud provider plugin will it have a node name that is the private dns name. For clusters that do not run with the AWS cloud provider plugin, the node name does not include the `.us-west-1.compute.internal` portion of the private dns name.

Additionally, this PR makes the docker image smaller by ~75% by using multi-stage builds.

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -13,105 +14,110 @@ import (
...
-func (n Node) PrivateHostname() (string, error) {
-	if n.hostname != "" {
-		return n.hostname, nil
-	}
-
-	i, err := n.instance()
-	if err != nil {
-		log.Error(err.Error())
-		return "", err
+func name(i *ec2.Instance, short bool) (string, error) {
```
Comment (andrewrynhard) [inline pkg/node/node.go]: Most of the changes boil down to allowing this function return the short version if specified in the config.

File: pkg/consumer/consumer.go [hunk 2]
```diff
@@ -74,15 +75,15 @@ func (c Consumer) Start(ctx context.Context) error {
...
-
-					err = n.Delete()
-					if err != nil {
-						log.Info(err.Error())
-					}
 				} else {
 					log.Warnf("Instance %s has already been terminated. Deleting...", event.EC2InstanceID)
 				}
 
+				err = n.Delete()
```
Comment (andrewrynhard) [inline pkg/consumer/consumer.go]: We should always attempt to delete, if the node has already been terminated.

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,13 @@
-FROM golang:1.8-onbuild
+FROM golang:1.8.7-alpine3.6 AS build
```
Comment (andrewrynhard) [inline Dockerfile]: Dockerfile had to be updated in order to support using the vendor dir.

Comment (woz5999) [reply Dockerfile]: out of curiosity, was there a particular package that was causing version issues?

Comment (andrewrynhard) [reply Dockerfile]: There was not, but I figured I would add vendor now to help prevent that in the future.

**Comment by woz5999** [comment]
lgtm

**Pull request merged**